docs/mp/guides/migration_3x.adoc

```diff
@@ -0,0 +1,172 @@
+///////////////////////////////////////////////////////////////////////////////
+
+    Copyright (c) 2022 Oracle and/or its affiliates.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+= Helidon MP 3.x Upgrade Guide
+:description: Helidon MP 3.x Upgrade Guide
+:keywords: helidon, porting, migration, upgrade, incompatibilities, 3.x
+:rootdir: {docdir}/../..
+
+include::{rootdir}/includes/mp.adoc[]
+
+
+In Helidon 3.x we have made some changes to APIs and runtime behavior. This guide
+will help you upgrade a Helidon MP 2.x application to 3.x.
+
+
+== Java 17 Runtime
+
+Java 11 is no longer supported in Helidon 3. Java 17 or newer is required. Please follow the instructions in xref:{rootdir}/includes/prerequisites.adoc}[Prerequisites] for proper installation.
+
+
+== javax.* namespace to jakarta.* namespace
+
+Helidon 3 supports MicroProfile 5 and _partially_ Jakarta EE 9.1 APIs. In Jakarta EE 9.1 the Java package namespace was changed from `javax` to `jakarta`. Therefore, you must change your application to use `jakarta` instead of `javax` for Jakarta EE packages.
+
+In version 3.x Helidon supports MicroProfile 5 specification, which now is fully migrated to `jakarta` namespace.
```

Comment:
5.0

Reply:
according to comment below it should be 5.1?

Reply:
fixed